hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/Histogram1D.java

```diff
@@ -8,61 +8,73 @@
 
 /**
  * One Dimensional histogram. Does not contain the column and membershipMap
- * todo: work in progress
+ * todo: Add createApproxHistogram that takes error parameters and uses sampling
  */
-public class Histogram1D implements IHistogram {
-   // final private IColumn column;
-   // private IMembershipSet membershipSet;
-   // final private double[] boundaries;
-    private final Bucket[] buckets;
-   // final private IStringConverter converter;
-    private int missingData;
-    private int outOfRange;
-    final private BucketSetDescription bucketDescription;
+public class Histogram1D {
 
-    public Histogram1D(@NonNull final BucketSetDescription bucketDescription) {
-        this.bucketDescription = bucketDescription;
-        this.buckets = new Bucket[bucketDescription.numOfBuckets];
-        createBuckets();
-    }
+    private final Bucket1D[] buckets;
 
-    private void createBuckets() {
-        for (int i = 0; i < this.bucketDescription.numOfBuckets; i++) {
-            this.buckets[i] = new Bucket(this.bucketDescription.boundaries[i], true,
-                    this.bucketDescription.boundaries[i + 1],
-                    (i == (this.bucketDescription.numOfBuckets - 1)));
-        }
-    }
+    private long missingData;
 
+    private long outOfRange;
 
-    public Histogram1D(BucketSetDescription bucketdescription, @NonNull final IColumn column,
-                       @NonNull final IMembershipSet membershipSet,
-                       @NonNull final IStringConverter converter) {
-        this.bucketDescription = bucketdescription;
-        this.buckets = new Bucket[bucketdescription.numOfBuckets]; //todo: create Buckets
-        this.createHistogram(column, membershipSet, converter);
+    private final IBucketsDescription1D bucketDescription;
+
+    public Histogram1D(final @NonNull IBucketsDescription1D bucketDescription) {
+        this.bucketDescription = bucketDescription;
+        this.buckets = new Bucket1D[bucketDescription.getNumOfBuckets()];
     }
 
     /**
-     * Creates the histogram explicitly and in full.
+     * Creates the histogram explicitly and in full. Should be called at most once.
      */
-    private void createHistogram(final IColumn column, final IMembershipSet membershipSet
+    public void createHistogram(final IColumn column, final IMembershipSet membershipSet
                                 ,final IStringConverter converter ) {
+        if (this.buckets[0] != null) //a histogram had already been created
+            throw new IllegalAccessError("A histogram cannot be created twice");
+        for (int i = 0; i < this.bucketDescription.getNumOfBuckets(); i++)
+            this.buckets[i] = new Bucket1D();
```

Comment:
The bucket does not have boundaries - you should probably use the new constructor I suggested.
I would move this code to the constructor.
In this way the constructor returns the "zero" for the monoid of histograms.